jlexer/lexer.go

```diff
@@ -46,7 +49,8 @@ type Lexer struct {
 	firstElement bool // Whether current element is the first in array or an object.
 	wantSep      byte // A comma or a colon character, which need to occur before a token.
 
-	err error // Error encountered during lexing, if any.
+	fatalError     error         // Fatal error occured during lexing. It is usually a syntax error.
+	SemanticErrors []*LexerError // Semantic errors occured during lexing. Marshalling will be continued after finding this errors.
```

Comment:
I'm not sure this needs to be exported. I'd probably add an accessor method, just as with the 'fatalError' field.